src/plonkish/compiler/cell_manager.rs

```diff
@@ -515,4 +576,102 @@ mod tests {
             1
         );
     }
+
+    #[test]
+    fn test_max_width_cm_different_columns() {
+        let mut unit = CompilationUnit::<()> {
+            forward_signals: vec![
+                ForwardSignal::new_with_phase(0, "forward signal 1".to_string()),
+                ForwardSignal::new_with_phase(0, "forward signal 2".to_string()),
+            ],
+            fixed_signals: vec![
+                FixedSignal::new_with_id(0, "fixed signal 1".to_string()),
+                FixedSignal::new_with_id(0, "fixed signal 2".to_string()),
+                FixedSignal::new_with_id(0, "fixed signal 3".to_string()),
+            ],
+            ..Default::default()
+        };
+
+        let mut step1 = StepType::new(1500, "step1".to_string());
+        step1.add_signal("c1");
+        step1.add_signal("d");
+        step1.add_signal("e");
+        let mut step2 = StepType::new(1501, "step2".to_string());
+        step2.add_signal("c2");
+
+        let step1 = Rc::new(step1);
+        let step2 = Rc::new(step2);
+
+        unit.step_types.insert(1500, Rc::clone(&step1));
+        unit.step_types.insert(1501, Rc::clone(&step2));
+
+        // forward signals: a, b; step1 internal: c1, d, e; step2 internal c2
+
+        let cm = MaxWidthCellManager {
```

Comment:
@atsupontio you should this test with different combinations of   max_width_advice and  max_width_fixed.

Including corner cases, like 0 and 1.

Reply:
If there are some repetation in the code use helper functions in the test module